src/components/VideosList/VideosList.test.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import VideosList from './VideosList.component';
+import data from '../../mocks/youtube-videos-mock.json';
+
+describe('VideosList', () => {
+  it('exists', () => {
+    render(React.createElement(VideosList, { videos: [] }));
+  });
+
+  it('renders message when there are no videos', () => {
+    render(React.createElement(VideosList, { videos: [] }));
+    const textElement = screen.getByText(/No videos found/i);
+    expect(textElement).toBeInTheDocument();
+  });
+
+  it('renders videos list', () => {
+    const { getAllByTestId } = render(
+      React.createElement(VideosList, { videos: data.items })
+    );
+    setTimeout(() => {
```

Comment:
When using `setTimeout` in testing, make sure to mock the timers behavior with `jest.useFakeTimers`, `jest.advanceTimersByTime` and `jest.useRealTimers`, so that you don't have to actually wait for the required time. Check:

https://jestjs.io/docs/timer-mocks

Let me know if you need help with this. I'm assuming you want to test the async behavior for `fetch` but I'm not sure.

Reply:
If I do not put the timeout the HTML does not render correctly, is there any other way to get it to finish rendering without the timeout?

Reply:
I did a couple of test and actually this is not running at all. The test ends before the `setTimeout` finishes. I suggest you to remove the setTimeout, and add a couple of console.logs to display what are you getting from `renderedTitles` and from `mockTitles`. We can sync tomorrow if you need further assistance.